src/main/java/com/mraof/minestuck/entity/animation/MobAnimation.java

```diff
@@ -0,0 +1,107 @@
+package com.mraof.minestuck.entity.animation;
+
+import net.minecraft.nbt.CompoundTag;
+
+/**
+ * Keeps track of a given animation. Can store stages of the animation in which to perform additional actions/events
+ */
+public class MobAnimation
+{
+	public static final int LOOPING_ANIMATION = -1; //when set to -1, it will continue looping until another animation overrides it
+	public static final MobAnimation.Actions IDLE_ACTION = Actions.IDLE;
+	public static final MobAnimation DEFAULT_IDLE_ANIMATION = new MobAnimation(IDLE_ACTION, LOOPING_ANIMATION, false, false);
+	
+	private final Actions actions;
+	private final int animationLength;
+	private final boolean freezeMovement;
+	private final boolean freezeSight;
+	
+	/**
+	 * @param actions category of the animation
+	 * @param animationLength last frame of the animation
+	 * @param freezeMovement whether the entity can move voluntarily during an animation, handled both in Goals and is updated upon the change of animation in AnimatedPathfinderMob
+	 * @param freezeSight whether the entity has their facing direction locked, handled by Goals
+	 */
+	public MobAnimation(Actions actions, int animationLength, boolean freezeMovement, boolean freezeSight)
+	{
+		this.actions = actions;
+		this.animationLength = animationLength;
+		this.freezeMovement = freezeMovement;
+		this.freezeSight = freezeSight;
+	}
+	
+	public Actions getAction()
+	{
+		return actions;
+	}
+	
+	public int getAnimationLength()
+	{
+		return animationLength;
+	}
+	
+	public boolean isLoopingAnimation()
+	{
+		return animationLength == LOOPING_ANIMATION;
+	}
+	
+	public boolean freezesMovement()
+	{
+		return freezeMovement;
+	}
+	
+	public boolean freezesSight()
+	{
+		return freezeSight;
+	}
+	
+	public CompoundTag getCompoundTag()
+	{
+		CompoundTag nbt = new CompoundTag();
+		nbt.putInt("actions", actions.ordinal());
+		nbt.putInt("animationLength", animationLength);
+		nbt.putBoolean("freezeMovement", freezeMovement);
+		nbt.putBoolean("freezeSight", freezeSight);
+		return nbt;
+	}
+	
+	public static MobAnimation createTrackerFromCompoundTag(CompoundTag nbtIn)
+	{
+		Actions actions = Actions.fromInt(nbtIn.getInt("actions"));
+		int animationLength = nbtIn.getInt("animationLength");
+		boolean freezeMovement = nbtIn.getBoolean("freezeMovement");
+		boolean freezeSight = nbtIn.getBoolean("freezeSight");
+		return new MobAnimation(actions, animationLength, freezeMovement, freezeSight);
+	}
```

Comment:
Consider whether these may or may not be used in the future, and remove them if they might not be.

Reply:
Tested in runServer and it didnt seem to make a difference when removed. Can be added back in later if needed